src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DockerImageDownloader.java

```diff
@@ -203,7 +206,23 @@ private File performDownloadSteps(Image image) throws PackageDownloadException,
             // Docker pull
             run(() -> {
                 if (credentialsUsable(image)) {
-                    dockerClient.pullImage(image);
+                    try {
+                        RetryUtils.runWithRetry(infiniteAttemptsRetryConfig, () -> {
+                            try {
+                                dockerClient.pullImage(image);
+                            } catch (ConnectionException e) {
+                                throw e;
+                            } catch (DockerServiceUnavailableException | InvalidImageOrAccessDeniedException
+                                     | UserNotAuthorizedForDockerException | DockerPullException e) {
+                                return null;
+                            }
+                            return null;
+                        }, "get-ecr-image", logger);
+                    } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)